iowrappers/redis_client.go

```diff
@@ -183,49 +177,70 @@ func (r *RedisClient) StorePlacesForLocation(context context.Context, geocodeInS
 	client := r.client
 	latLng, _ := utils.ParseLocation(geocodeInString)
 	lat, lng := latLng[0], latLng[1]
-	wg := &sync.WaitGroup{}
-	wg.Add(len(places))
 	for _, place := range places {
 		sortedSetKey := strings.Join([]string{geocodeInString, string(POI.GetPlaceCategory(place.LocationType))}, "_")
 		dist := utils.HaversineDist([]float64{lat, lng}, []float64{place.GetLocation().Latitude, place.GetLocation().Longitude})
 		_, err := client.ZAdd(context, sortedSetKey, redis.Z{Score: dist, Member: place.ID}).Result()
 		if err != nil {
 			return err
 		}
-		r.setPlace(context, place, wg)
+		if err := r.setPlace(context, place); err != nil {
+			return err
+		}
 	}
-	wg.Wait()
 	return nil
 }
 
 // SetPlacesAddGeoLocations stores two types of information in redis
 // 1. key-value pair, {placeID: POI.place}
 // 2. add place to the correct bucket in geohashing for nearby search
+// Batches places into groups to reduce goroutine overhead
 func (r *RedisClient) SetPlacesAddGeoLocations(c context.Context, places []POI.Place) {
+	if len(places) == 0 {
+		return
+	}
+
+	const batchSize = 100 // Process 100 places per pipeline
+	numBatches := (len(places) + batchSize - 1) / batchSize
+
```

Comment:
what happens with increased batch size ? Will that affect the backend compute in anyway ?